src/core/store/modules/vaults/vaults.actions.ts

```diff
@@ -343,6 +343,67 @@ const withdrawVault = createAsyncThunk<
   }
 );
 
+const withdrawAllVault = createAsyncThunk<
+  void,
+  { vaultAddress: string; targetTokenAddress: string; slippageTolerance?: number },
+  ThunkAPI
+>(
+  'vaults/withdrawAllVault',
```

Comment:
if you set this as `vaults/withdrawVault` then there is no need to have a different reducer cases :)

Reply:
Will refactor to a new approach of reusing the withdraw action